samples/net/sockets/mt_dumb_http_server/src/main.c

```diff
@@ -0,0 +1,355 @@
+
+#if defined(CONFIG_NET_IPV4)
+K_THREAD_STACK_ARRAY_DEFINE(tcp4_handler_stack, CONFIG_NET_SAMPLE_NUM_HANDLERS,
+			    STACK_SIZE);
+static struct k_thread tcp4_handler_thread[CONFIG_NET_SAMPLE_NUM_HANDLERS];
+static k_tid_t tcp4_handler_tid[CONFIG_NET_SAMPLE_NUM_HANDLERS];
+static bool tcp4_handler_in_use[CONFIG_NET_SAMPLE_NUM_HANDLERS];
+#endif
+
+#if defined(CONFIG_NET_IPV6)
+K_THREAD_STACK_ARRAY_DEFINE(tcp6_handler_stack, CONFIG_NET_SAMPLE_NUM_HANDLERS,
+			    STACK_SIZE);
+static struct k_thread tcp6_handler_thread[CONFIG_NET_SAMPLE_NUM_HANDLERS];
+static k_tid_t tcp6_handler_tid[CONFIG_NET_SAMPLE_NUM_HANDLERS];
+static bool tcp6_handler_in_use[CONFIG_NET_SAMPLE_NUM_HANDLERS];
+#endif
+
+static int tcp4_sock;
+static int tcp4_accepted[CONFIG_NET_SAMPLE_NUM_HANDLERS];
+static int tcp6_sock;
+static int tcp6_accepted[CONFIG_NET_SAMPLE_NUM_HANDLERS];
+
+static void process_tcp4(void);
+static void process_tcp6(void);
+
+K_THREAD_DEFINE(tcp4_thread_id, STACK_SIZE,
+		process_tcp4, NULL, NULL, NULL,
+		THREAD_PRIORITY, 0, K_FOREVER);
+
+K_THREAD_DEFINE(tcp6_thread_id, STACK_SIZE,
+		process_tcp6, NULL, NULL, NULL,
+		THREAD_PRIORITY, 0, K_FOREVER);
+
+static ssize_t sendall(int sock, const void *buf, size_t len)
+{
+	while (len) {
+		ssize_t out_len = send(sock, buf, len, 0);
+
+		if (out_len < 0) {
+			return out_len;
+		}
+		buf = (const char *)buf + out_len;
+		len -= out_len;
+	}
+
+	return 0;
+}
+
+static int setup(int *sock, struct sockaddr *bind_addr,
+		 socklen_t bind_addrlen)
+{
+	int ret;
+
+#if defined(CONFIG_NET_SOCKETS_SOCKOPT_TLS)
+	*sock = socket(bind_addr->sa_family, SOCK_STREAM, IPPROTO_TLS_1_2);
+#else
+	*sock = socket(bind_addr->sa_family, SOCK_STREAM, IPPROTO_TCP);
+#endif
+	if (*sock < 0) {
+		LOG_ERR("Failed to create TCP socket: %d", errno);
+		return -errno;
+	}
+
+#if defined(CONFIG_NET_SOCKETS_SOCKOPT_TLS)
+	sec_tag_t sec_tag_list[] = {
+		SERVER_CERTIFICATE_TAG,
+	};
+
+	ret = setsockopt(*sock, SOL_TLS, TLS_SEC_TAG_LIST,
+			 sec_tag_list, sizeof(sec_tag_list));
+	if (ret < 0) {
+		LOG_ERR("Failed to set TCP secure option %d", errno);
+		ret = -errno;
+	}
+#endif
+
+	ret = bind(*sock, bind_addr, bind_addrlen);
+	if (ret < 0) {
+		LOG_ERR("Failed to bind TCP socket %d", errno);
+		return -errno;
+	}
+
+	ret = listen(*sock, MAX_CLIENT_QUEUE);
+	if (ret < 0) {
+		LOG_ERR("Failed to listen on TCP socket %d", errno);
+		ret = -errno;
+	}
+
+	return ret;
+}
+
+static void handle_data(void *ptr1, void *ptr2, void *ptr3)
+{
+	ARG_UNUSED(ptr1);
+	int *sock = ptr2;
+	bool *in_use = ptr3;
+	int client;
+	int received;
+	int ret;
+	char buf[100];
+
+	client = *sock;
+
+	do {
+		received = recv(client, buf, sizeof(buf), 0);
+		if (received == 0) {
+			/* Connection closed */
+			LOG_DBG("[%d] Connection closed by peer", client);
+			ret = 0;
+			break;
+		} else if (received < 0) {
+			/* Socket error */
+			ret = -errno;
+			LOG_ERR("[%d] Connection error %d", client, ret);
+			break;
+		}
+
+		if (strstr(buf, "\r\n\r\n")) {
```

Comment:
Agreed, the check is very dumb. Just did it like this in order to avoid spending too much time with this. I will add comment saying it is not to be used in real applications.